lib/events/complete.go

```diff
@@ -161,6 +171,40 @@ func (u *UploadCompleter) Serve(ctx context.Context) error {
 	}
 }
 
+func (u *UploadCompleter) PerformPeriodicCheck(ctx context.Context) {
+	// If configured with a server ID, then acquire a semaphore prior to completing uploads.
+	// This is used for auth's upload completer and ensures that multiple auth servers do not
+	// attempt to complete the same uploads at the same time.
+	if u.cfg.Semaphores != nil {
+		u.log.Infof("%v: acquiring semaphore in order to complete uploads", u.cfg.ServerID)
+
+		ctx, cancel := context.WithCancel(ctx)
+		defer cancel()
+
+		_, err := services.AcquireSemaphoreLock(ctx, services.SemaphoreLockConfig{
+			Service: u.cfg.Semaphores,
+			Clock:   u.cfg.Clock,
+			Expiry:  u.cfg.CheckPeriod - 1*time.Minute,
```

Comment:
nit: hard-coded subtraction is brittle and the default check period is fairly long, how about half the check period rounded up instead?  (e.g. `(u.cfg.CheckPeriod / 2) + 1`).

Reply:
Had a feeling you'd suggest something different here 😄